octoprint_mrbeamdoc/__init__.py

```diff
@@ -1,22 +1,191 @@
+
+    return mrbeamdoc
+
+
+def get_version():
+    return __version__
+
+
+class MrBeamDoc:
+    def __init__(self, definition, language, extension='pdf'):
+        self.definition = definition
+        self.language = language
+        self.extension = extension
+
+    def __repr__(self):
+        return 'MrBeamDoc(definition=%(definition)s, language=%(language)s, extension=%(extension)s)' % {
+            'definition': repr(self.definition),
+            'language': self.language.name,
+            'extension': self.extension,
+        }
+
+    def get_folder(self):
+        return os.path.join(_ROOT, 'docs')
+
+    def get_file_name_with_extension(self):
+        return '%(mrbeamdoc_type)s_%(mrbeam_model)s_%(language)s.%(extension)s' % {
+            'mrbeamdoc_type': self.definition.mrbeamdoc_type.value,
+            'mrbeam_model': self.definition.mrbeam_model.value,
+            'language': self.language.value,
+            "extension": self.extension}
+
+    def get_file_reference(self):
+        return os.path.join(self.get_folder(), self.get_file_name_with_extension())
+
+
+class SupportedLanguage(Enum):
+    GERMAN = 'de'
+    ENGLISH = 'en'
+    SPANISH = 'es'
+    FRENCH = 'fr'
+    ITALIAN = 'it'
+    DUTCH = 'nl'
+    FINNISH = 'fi'
+
+
+class MrBeamDocType(Enum):
+    USER_MANUAL = 'UserManual'
+    QUICKSTART_GUIDE = 'QuickstartGuide'
+
+
+class MrBeamModel(Enum):
+    MRBEAM2 = 'mrbeam2'
+    MRBEAM2_DC_R1 = 'mrbeam2_dc_r1'
+    MRBEAM2_DC_R2 = "mrbeam2_dc_r2"
+    DREAMCUT_ALIAS = 'dreamcut'
+    DREAMCUT = 'mrbeam2_dc'
+    DREAMCUT_S = "mrbeam2_dc_s"
+
+
+class MrBeamDocDefinition:
+    def __init__(self, mrbeamdoc_type, mrbeam_model, supported_languages, additionally_valid_models=[]):
+        self.mrbeamdoc_type = mrbeamdoc_type
+        self.mrbeam_model = mrbeam_model
+        additionally_valid_models.append(mrbeam_model)
+        self.valid_models = set(item for item in additionally_valid_models)
+        self.supported_languages = set(item for item in supported_languages)
+
+    def __repr__(self):
+        return """MrBeamDocDefinition(mrbeamdoc_type=%(mrbeamdoc_type)s,
+            mrbeam_model=%(mrbeam_model)s,
+            valid_models=%(valid_models)s,
+            supported_languages=%(supported_languages)s
+        )""" % {
+            'mrbeamdoc_type': self.mrbeamdoc_type.name,
+            'mrbeam_model': self.mrbeam_model.name,
+            'valid_models': ','.join(item.name for item in self.valid_models),
+            'supported_languages': ','.join(item.name for item in self.supported_languages)
+        }
+
+    def is_valid_for_model(self, mrbeam_model):
+        return mrbeam_model in self.valid_models
+
+    def is_language_supported(self, language):
+        return language in self.supported_languages
+
+    def is_valid(self, mrbeamdoc_type, mrbeam_model, language):
+        return self.mrbeamdoc_type == mrbeamdoc_type and self.is_valid_for_model(
```

Comment:
you should surround checks with () so that it can be easily readable and to make no chance for making a mistake

Reply:
What do you mean exactly? the whole return statement? or the equality? or both?

Reply:
self.mrbeamdoc_type == 
mrbeamdoc_type and
 self.is_valid_for_model( mrbeam_model) and
 self.is_language_supported(
            language)

This part

Reply:
Let's keep the current naming for now.